### PR TITLE
[3/5] remove pixel detector base, add to spot detector, move spots up one level

### PR DESCRIPTION
**Pull request by ambrosejcarr**
- Remove pixel detector base
- Move pixel detector to spots (the bases were identical)
- Remove `features` (move `spots` up one level)

### CODE REVIEW

File: starfish/pipeline/spots/detector/_base.py [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
 import numpy as np
 import xarray as xr
 
 from starfish._stack import ImageStack
 from starfish.intensity_table import IntensityTable
 from starfish.pipeline.algorithmbase import AlgorithmBase
-from starfish.pipeline.features.spot_attributes import SpotAttributes
+from starfish.pipeline.spots.spot_attributes import SpotAttributes
+from starfish.pipeline.spots.detector.combine_adjacent_features import (
```
Comment (ttung) [inline starfish/pipeline/spots/detector/_base.py]: sort, and i don't think you need the multiline import.

File: starfish/pipeline/spots/detector/_base.py [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
 from starfish.types import Number
 
 
 class SpotFinderAlgorithmBase(AlgorithmBase):
     def find(
             self,
             hybridization_image: ImageStack,
-            blobs_image: Union[np.ndarray, xr.DataArray]
-    ) -> IntensityTable:
+    ) -> Union[IntensityTable, Tuple[IntensityTable, ConnectedComponentDecodingResult]]:
```
Comment (ttung) [inline starfish/pipeline/spots/detector/_base.py]: WHOA what?  this seems really bad.  is this an intermediate step before we normalize?

File: notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.ipynb [hunk 3]
```diff
@@ -152,7 +152,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "from starfish.image._filter import DeconvolvePSF\n",
+    "from starfish.image._filter.richardson_lucy_deconvolution import DeconvolvePSF\n",
```
Comment (ttung) [inline notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.ipynb]: hidden stuff!

File: starfish/pipeline/spots/detector/_base.py (hunk 2 above)
Comment (ambrosejcarr) [reply starfish/pipeline/spots/detector/_base.py]: I hope so. Deep and I need to talk about how the PixelSpotFinder should work.

**Pull request merged**